src/Aspire.Dashboard/Extensions/ResourceViewModelExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Dashboard.Model;
+
+namespace Aspire.Dashboard.Extensions;
+
+public static class ResourceViewModelExtensions
```

Comment:
All these methods could be extensions of KnownResourceState rather than the view model.

Reply:
You're absolutely right, the reason I kept it on ResourceViewModel was because putting an extension on KnownResourceState results in some additional code because of the nullability of `KnownState`. I'd like to keep it for now, but I'd be happy to chat offline about this too!

ie: `r.Value.KnownState?.IsHiddenState() is false` instead of `!r.Value.IsHiddenState()`